app/controllers/hyrax/generic_works_controller.rb

```diff
@@ -68,9 +81,17 @@ def download
 
   def assign_visibility
     if params["isDraft"] == Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
-     params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
+      params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
     else
       params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC
+      if params['action'] == 'update'
```

Comment:
Ideally this would use action_name method instead of params[:action] or params['action'].  Please let me know if that should be fixed.